tests/test_raw_node.rs

```diff
@@ -141,6 +141,56 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_read_index ensures that RawNode.read_index sends the MsgReadIndex message
+// to the underlying raft. It also ensures that ReadState can be read out.
+#[test]
+fn test_raw_node_read_index() {
```

Comment:
No, we should use the same test cases as etcd/raft. I think we can change `allow_step` in `raft` to a function.